lib/tarfiles.py

```diff
@@ -82,29 +88,30 @@ def do_tarballs(args):
         if tfn.startswith("dropbox:"):
             # move it to dropbox area, pretend they gave us plain path
 
-            if args.use_dropbox == "cvmfs" or args.use_dropbox == None:
+            if args.use_dropbox == "cvmfs" or args.use_dropbox is None:
                 digest, tf = slurp_file(tfn[8:])
                 proxy, token = get_creds()
 
-                if token:
-                    f = open(token,'r')
-                    tokenstr = f.read()
-                    f.close()
-
                 if not args.group:
                     raise ValueError("No --group specified!")
-                cid = "".join((args.group, "%2F", digest))
-                location = pubapi_exists(cid, proxy) # tokenstr=tokenstr
-                if not location:
-                    pubapi_publish(cid, tf, proxy) # tokenstr=tokenstr
-                    for i in range(20):
-                        time.sleep(30)
-                        location = pubapi_exists(cid, proxy) # tokenstr=tokenstr
-                        if location:
+
+                # cid looks something like dune/bf6a15b4238b72f82...(long hash)
+                cid = f"{args.group}/{digest}"
+                if args.verbose:
+                    print(f"Using RCDS to publish tarball\ncid: {cid}")
+
+                publisher = TarfilePublisherHandler(cid, proxy, token)
+                location = publisher.cid_exists()
+                if location is None:
+                    publisher.publish(tf)
```

Comment:
What happens if `publish()` fails? There should be some retry logic, either here or in the `TarfilePublisherHandler` methods. Maybe the latter is preferable? There's already the `pubapi_operation` wrapper so could add retry logic there. Need to be careful about nested retries though.

Reply:
Talked to Kevin offline, and he suggested putting retry logic into the wrapper in `TarfilePublisherHandler`.

Reply:
Done.  Currently, there are nested retries, but once we do #38 , that won't be an issue.

Reply:
Still need to handle if all retries of `publish()` fail, and stop the submission.

Reply:
Done in the TarfilePublisherHandler wrapper.  I applied it to all the RCDS calls.